File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        walls = 0
        rows = len(grid)
        cols = len(grid[0])

        for i in range(rows):
            for j in range(cols):
                if grid[i][j] == 1:
                    walls += 4
                else:
                    continue
                if i+1 < rows and grid[i+1][j]:
                    walls -= 1
                if i-1 >= 0 and grid[i-1][j]:
                    walls -= 1
                if j+1 < cols and grid[i][j+1]:
                    walls -=1
                if j-1 >= 0 and grid[i][j-1]:
                    walls -=1
        return walls

File: 0463-island-perimeter/0463-island-perimeter.c
int islandPerimeter(int** grid, int gridSize, int* gridColSize) {
    int walls = 0;
    for (int i = 0; i < gridSize; i++) {
        for (int j = 0; j < gridColSize[i]; j++) {
            if (grid[i][j] == 1) {
                walls += 4;
            }
            else continue;
            if ((i+1 < gridSize) && (grid[i+1][j])) {
                walls--;
            }
            if ((i-1 >= 0) && (grid[i-1][j])) {
                walls--;
            }
            if ((j+1 < gridColSize[i]) && (grid[i][j+1])) {
                walls--;
            }
            if ((j-1 >= 0) && (grid[i][j-1])) {
                walls--;
            }
        }     
    }
    return walls;
}